### PR TITLE
fix: Use custom controls for iOS video (#7729)

### LINKED ISSUE #7729: [Bug]: Metamask crashes on the Protect your wallet flow on iOS 17.2 beta
### Describe the bug

The app crashes on version 7.9.1 on mobile when users go through the Protect your wallet flow to set a password and back up their SRP.

Edit from: @Cal-L @hesterbruikman 
Navigating to another screen after going full screen on a video for iOS 17 causes a crash in the app. It has been reported here https://github.com/react-native-video/react-native-video/issues/3329. Based on the logs provided by the MM user and the issue report, the theory aligns.

### Expected behavior

Users should be able to go through the Protect your wallet flow and back up their SRP normally without the application closing.

### Screenshots/Recordings

Original thread:  https://consensys.slack.com/archives/C8RSKCNCD/p1699369438003369

### Steps to reproduce

1. Create a wallet on an iPhone using iOS 17.2 beta (the only one confirmed at this time)
2. Skip the SRP backup
3. Add funds
4. You will be prompted to the **Protect your wallet** flow
5. After watching the video and going to the password screen, the crash happens

### Error messages or log output

```shell
None
```

### Version

7.9.1

### Build type

None

### Device

iPhone on iOS 17.2 beta

### Operating system

iOS, Other (please elaborate in the "Additional Context" section)

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by Cal-L**
## **Description**

It was discovered that the video begins crashing the app for iOS 17.2+ physical devices. While there is no solution in the latest react-native-video package at the moment (https://github.com/react-native-video/react-native-video/issues/3329), we can patch it by removing the `controls` prop and temporarily use custom controls (a play/pause + mute controls). The tradeoff is that other features such as full screen and seeking will not be available on iOS.

## **Related issues**

Fixes: #7729

## **Manual testing steps**

1. Go to this page...
2.
3.

### **After**
Screenshot 2023-11-09 at 9 19 57 AM

iOS 17.2 Interaction
https://github.com/MetaMask/metamask-mobile/assets/10508597/8976d5b7-d276-4834-9496-0ba8608e25e3

Video with play/pause + mute controls on settings
https://github.com/MetaMask/metamask-mobile/assets/10508597/492fb78d-3703-4714-bed1-8ddd7b631efa

Video with play/pause + mute controls on onboarding
https://github.com/MetaMask/metamask-mobile/assets/10508597/14780058-c595-4fb9-8aab-9987120c8126

Android remains the same
https://recordit.co/pTuloJmuIn

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/Views/MediaPlayer/index.js [hunk 1]
```diff
@@ -44,17 +114,50 @@ function MediaPlayer({ uri, style, onClose, textTracks, selectedTextTrack }) {
...
+                  name={isPlaying ? 'ios-play' : 'ios-pause'}
+                  size={24}
+                  color={colors.overlay.inverse}
+                  style={styles.playIcon}
+                />
+              </View>
+            </Animated.View>
+          </TapGestureHandler>
+          {isPlaying ? (
+            <TouchableOpacity
```
Comment (tommasini) [inline app/components/Views/MediaPlayer/index.js:148]: It's possible to use the Button of component library, the Icon variant?

**Review by tommasini — APPROVED** [review]
Awesome work! LGTM, it would be cool to convert this to typescript and do an unit test for when the mute button is displayed if you have the time since this is a sev1

File: app/components/Views/MediaPlayer/index.js (hunk 1 above)
Comment (Cal-L) [reply app/components/Views/MediaPlayer/index.js:148]: Unfortunately, we don't have play and pause icons at the moment. This is also intended to be a temporarily solution until the react native video bug is resolved in the package itself.

**Comment by chrisleewilcox** [comment]
Was not able to crash app on iPhone 12 iOS 16.2
https://recordit.co/4J2OvXKiau

**Review by chrisleewilcox — APPROVED** [review]
LGTM

**Pull request merged**